include/message_data_fmt.h

```diff
@@ -1,180 +1,187 @@
-#define CTRL_ADJUSTABLE 2
-#define CTRL_BLUE       3
-#define CTRL_LIGHTBLUE  4
-#define CTRL_PURPLE     5
-#define CTRL_YELLOW     6
-#define CTRL_BLACK      7
+typedef enum {
+    TEXTBOX_BG_X_LEFT,
+    TEXTBOX_BG_X_RIGHT
+} TextboxBackgroundIndex;
+
+typedef enum {
+    TEXTBOX_BG_FGCOL_WHITE,
+    TEXTBOX_BG_FGCOL_DARK_RED,
+    TEXTBOX_BG_FGCOL_ORANGE,
+    TEXTBOX_BG_FGCOL_WHITE_3,
+    TEXTBOX_BG_FGCOL_WHITE_4,
+    TEXTBOX_BG_FGCOL_WHITE_5,
+    TEXTBOX_BG_FGCOL_WHITE_6,
+    TEXTBOX_BG_FGCOL_WHITE_7
+} TextboxBackgroundForegroundColor;
+
+typedef enum {
+    TEXTBOX_BG_BGCOL_BLACK,
+    TEXTBOX_BG_BGCOL_GOLD,
+    TEXTBOX_BG_BGCOL_BLACK_2,
+    TEXTBOX_BG_BGCOL_BLACK_3
+} TextboxBackgroundBackgroundColor;
+
+typedef enum {
+    TEXTBOX_BG_Y_OFFSET_1,
+    TEXTBOX_BG_Y_OFFSET_2
+} TextboxBackgroundYOffsetIndex;
 
 #ifdef MESSAGE_DATA_STATIC
 // For use in message_data_static files
 
-#define ARG(x) x
-
-// while a control character, newlines are handled in the charmap conversion
-// stage to allow normal newline \n usage in message_data_static files
-#define NEWLINE                 STR(CTRL_NEWLINE)
-#define END                     STR(CTRL_END)
-#define BOX_BREAK               STR(CTRL_BOX_BREAK)
-#define COLOR(x)                STR(CTRL_COLOR)                ARG(x) // 1
-#define SHIFT(x)                STR(CTRL_SHIFT)                ARG(x) // 1
-#define TEXTID(x)               STR(CTRL_TEXTID)               ARG(x) // 2
-#define QUICKTEXT_ENABLE        STR(CTRL_QUICKTEXT_ENABLE)
-#define QUICKTEXT_DISABLE       STR(CTRL_QUICKTEXT_DISABLE)
-#define PERSISTENT              STR(CTRL_PERSISTENT)
-#define EVENT                   STR(CTRL_EVENT)
-#define BOX_BREAK_DELAYED(x)    STR(CTRL_BOX_BREAK_DELAYED)    ARG(x) // 1
-#define AWAIT_BUTTON_PRESS      STR(CTRL_AWAIT_BUTTON_PRESS)
-#define FADE(x)                 STR(CTRL_FADE)                 ARG(x) // 1
-#define NAME                    STR(CTRL_NAME)
-#define OCARINA                 STR(CTRL_OCARINA)
-#define FADE2(x)                STR(CTRL_FADE2)                ARG(x) // 2
-#define SFX(x)                  STR(CTRL_SFX)                  ARG(x) // 2
-#define ITEM_ICON(x)            STR(CTRL_ITEM_ICON)            ARG(x) // 1
-#define TEXT_SPEED(x)           STR(CTRL_TEXT_SPEED)           ARG(x) // 1
-#define BACKGROUND(x,y,z)       STR(CTRL_BACKGROUND)           ARG(x) ARG(y) ARG(z)
-#define MARATHON_TIME           STR(CTRL_MARATHON_TIME)
-#define RACE_TIME               STR(CTRL_RACE_TIME)
-#define POINTS                  STR(CTRL_POINTS)
-#define TOKENS                  STR(CTRL_TOKENS)
-#define UNSKIPPABLE             STR(CTRL_UNSKIPPABLE)
-#define TWO_CHOICE              STR(CTRL_TWO_CHOICE)
-#define THREE_CHOICE            STR(CTRL_THREE_CHOICE)
-#define FISH_INFO               STR(CTRL_FISH_INFO)
-#define HIGHSCORE(x)            STR(CTRL_HIGHSCORE)            ARG(x) // 1
-#define TIME                    STR(CTRL_TIME)
+// Text is preprocessed with modern cpp so varargs is valid
```

Comment:
This is still a bit of a headscratcher IMO, maybe clarify that it's IDO in particular that doesn't support varags and "preprocessed with modern CPP" is something that we do specially for text in the Makefile?

Reply:
Tried to elaborate on this comment, what do you think now?